util/contentutil/fetcher.go

```diff
@@ -29,6 +30,17 @@ func (p *fetchedProvider) ReaderAt(ctx context.Context, desc ocispecs.Descriptor
 	return &readerAt{Reader: rc, Closer: rc, size: desc.Size}, nil
 }
 
+func (p *fetchedProvider) Info(ctx context.Context, dgst digest.Digest) (content.Info, error) {
+	ra, err := p.ReaderAt(ctx, ocispecs.Descriptor{Digest: dgst})
+	if err != nil {
+		return content.Info{}, err
+	}
+	size := ra.Size()
```

Comment:
This doesn't actually work. Line 30 you can see that this just returns the size passed into `ReaderAt` (zero in this case).

Is the `Info` really needed in this case? Eg. I see it used in `azblob` but in there we have the descriptor already.

Doing a full `GET` request just to get size should be avoided as well.